offa.csl

```diff
@@ -0,0 +1,438 @@
+        </names>
+      </else-if>
+      <else>
+        <text variable="title-short"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="title">
+    <choose>
+      <if variable="title" match="none">
+        <choose>
+          <if type="personal_communication" match="none">
+            <text variable="genre" text-case="capitalize-first"/>
+          </if>
+        </choose>
+      </if>
+      <else-if type="bill book graphic legal_case motion_picture report song thesis" match="any">
+        <text variable="title"/>
+        <text variable="volume" prefix=" "/>
+      </else-if>
+      <else>
+        <text variable="title"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="edition">
+    <choose>
+      <if type="bill book graphic legal_case motion_picture report song chapter paper-conference" match="any">
+        <choose>
+          <if is-numeric="edition">
+            <text variable="edition" vertical-align="sup"/>
+          </if>
+        </choose>
+      </if>
+    </choose>
+  </macro>
+  <macro name="locators">
+    <choose>
+      <if type="article-journal">
+        <number prefix=" " variable="volume"/>
+        <text variable="issue" prefix=", "/>
+      </if>
+    </choose>
+  </macro>
+  <macro name="locators-chapter">
+    <choose>
+      <if type="chapter paper-conference" match="any">
+        <text variable="page" prefix=" "/>
+      </if>
+    </choose>
+  </macro>
+  <macro name="locators-article">
+    <choose>
+      <if type="article-newspaper">
+        <group delimiter=", ">
+          <group delimiter=" ">
+            <text variable="edition" prefix=", "/>
+            <text term="edition" prefix=", "/>
+          </group>
+          <group>
+            <text term="section" form="short" suffix=" "/>
+            <text variable="section"/>
+          </group>
+        </group>
+      </if>
+      <else-if type="article-journal">
+        <text variable="page" prefix=", "/>
+      </else-if>
+    </choose>
+  </macro>
+  <macro name="date">
+    <date variable="issued">
+      <date-part name="year"/>
+    </date>
+  </macro>
+  <macro name="place-date">
+    <choose>
+      <if type="book thesis chapter paper-conference" match="any">
+        <choose>
+          <if match="none" variable="edition">
+            <text value=""/>
+          </if>
+        </choose>
+        <group delimiter="" prefix=" (" suffix=")">
+          <group delimiter=" ">
+            <choose>
+              <if type="thesis" match="any">
+                <text variable="genre" suffix=" "/>
+                <text variable="publisher"/>
+              </if>
+              <else>
+                <choose>
+                  <if match="any" variable="original-date original-publisher-place">
+                    <text variable="original-publisher-place"/>
+                    <date date-parts="year" form="text" variable="original-date" prefix=" "/>
+                    <text value="; "/>
```

Comment:
use a group with a suffix. Generally avoid `text value` for most purposes.

Reply:
Modifications here are difficult to test for me as somehow "original-date" and "original-publisher-place" are not valid fields in my Zotero-Bib.

Reply:
Presumably you mean Zotero, not Zotero-bib (i.e. zbib.org)? You can just add those to the extra field and they'll get picked up by Zotero (true for any CSL variable):
```
original-date: 1776
original-publisher: Glasgow University Press
```

Reply:
I did not know this. Very handy. Many thanks for pointing this out!